### DISCUSSION TITLE
useParams

### DISCUSSION THREAD

**Discussion by renanrodrigueszup** [vercel/next.js#44859] [Help]
From what I could learn from the migration guide to appDir, the useSearchParams would be the equivalent to router.query.
However, this new hook doesnt provide us the the route params. 
I noticed there's a undocumented client component hook called [useParams](https://beta.nextjs.org/docs/api-reference/use-params). I was wondering if this would be the hook providing us route params (i.e slugs) on the appDir structure ?

**Comment by JasperAlexander** [comment]
Is there already an answer to this (useParams)?

You can use `params: { [key: string]: string | string[] | undefined }` on client components.

**Comment by jon-w1** [comment]
+1 
I'm interested in the `useParams` hook instead of the `params` prop so I don't have to prop drill the dynamic path segments into child components.

**Comment by timneutkens** [comment]
Opened a PR for `useParams` here: https://github.com/vercel/next.js/pull/47373
